sg/path.go

```diff
@@ -24,6 +26,34 @@ func FromWorkDir(pathElems ...string) string {
 	return filepath.Join(append([]string{cwd}, pathElems...)...)
 }
 
+// FromOSCache reutrns the path relative to OS cache folder, and a boolean indicating
+// whether such folder exists or not.
+func FromOSCache(pathElems ...string) (string, bool) {
+	switch runtime.GOOS {
+	case "darwin":
+		return filepath.Join(append([]string{os.Getenv("HOME"), "Library", "Caches"}, pathElems...)...), true
+	case "linux":
+		if xdgCacheHome := os.Getenv("XDG_CACHE_HOME"); xdgCacheHome != "" {
+			return filepath.Join(append([]string{xdgCacheHome}, pathElems...)...), true
+		} else {
+			return filepath.Join(append([]string{os.Getenv("HOME"), ".cache"}, pathElems...)...), true
+		}
+	default:
+		return "", false
+	}
+}
```

Comment:
Is there a problem with [os.UserCacheDir](https://pkg.go.dev/os#UserCacheDir) ?

If so you should document it in a comment